schemas/HPXMLBaseElements.xsd

```diff
@@ -6230,6 +6395,56 @@
 		</xs:sequence>
 		<xs:attribute name="dataSource" type="DataSource"/>
 	</xs:complexType>
+	<xs:complexType name="VehicleBattery">
+		<xs:sequence>
+			<xs:element minOccurs="0" name="BatteryType" type="BatteryType"/>
+			<xs:element minOccurs="0" name="NominalCapacity" type="BatteryCapacityType" maxOccurs="unbounded">
+				<xs:annotation>
+					<xs:documentation>The total energy available when the battery is discharged starting from 100% state of charge until it reaches the cut-off voltage.</xs:documentation>
+				</xs:annotation>
+			</xs:element>
+			<xs:element minOccurs="0" name="UsableCapacity" type="BatteryCapacityType" maxOccurs="unbounded">
+				<xs:annotation>
+					<xs:documentation>The stored energy that can actually be used. In most cases usable capacity is less than the nominal capacity.</xs:documentation>
+				</xs:annotation>
+			</xs:element>
+			<xs:element minOccurs="0" name="CoolingStrategy" type="BatteryCoolingStrategy"/>
+			<xs:element minOccurs="0" name="RatedPowerOutput" type="Power">
+				<xs:annotation>
+					<xs:documentation>[W] The amount of power the battery typically generates under non-peak conditions.</xs:documentation>
+				</xs:annotation>
+			</xs:element>
+			<xs:element name="PeakPowerOutput" type="Power" minOccurs="0">
+				<xs:annotation>
+					<xs:documentation>[W] The peak power that the battery can generate for a short period of time.</xs:documentation>
+				</xs:annotation>
+			</xs:element>
+			<xs:element minOccurs="0" name="NominalVoltage" type="HPXMLDecimal">
+				<xs:annotation>
+					<xs:documentation>[V] The nominal voltage is the battery voltage when the state of charge is 0.5 (midway between being fully charged, and fully discharged)
+												with a 0.2C discharge current.</xs:documentation>
+				</xs:annotation>
+			</xs:element>
+			<xs:element minOccurs="0" name="RoundTripEfficiency" type="FractionExcludingZero">
+				<xs:annotation>
+					<xs:documentation>Not all the power that is used to charge the battery is available during discharge. Round trip efficiency is the ratio of the energy put
+												in to the energy retrieved from storage.</xs:documentation>
+				</xs:annotation>
+			</xs:element>
+			<xs:element minOccurs="0" name="GridConnected" type="HPXMLBoolean">
+				<xs:annotation>
+					<xs:documentation>Has the ability to feed electricity back on to the grid.</xs:documentation>
+				</xs:annotation>
+			</xs:element>
+			<xs:element minOccurs="0" name="BuildingConnected" type="HPXMLBoolean">
+				<xs:annotation>
+					<xs:documentation>Has the ability to feed electricity to the building.</xs:documentation>
+				</xs:annotation>
+			</xs:element>
```

Comment:
Are these really separate capabilities? An EV could have the ability to be grid connected but not building connected, or vice versa? I guess I assumed it was either connected, in which case it could do both, or neither.

Reply:
Also, I think these properties should be moved up from the `Battery` element to the `BEV` element. They are more a property of the car than the battery itself, right?

Reply:
Good question, I think from a technology standpoint they may not be separate, but there may be instances where a home has one and not the other, for example the utility does not offer a program for V2G.

I actually think bidirectional charging is dependent on both the vehicle and the charger, and you would need both to be compatible. Does it make sense to have similar `BuildingConnected` and `GridConnected` attributes for the BEV and the charger?

Reply:
It sounds like we need to be clear as to whether these fields describe capabilities or uses. E.g., a vehicle may have V2G capabilities, but it might not be used for a given home.

Reply:
What about adding a new type that looks like:

```
<xs:element minOccurs="0" name="BidirectionalCharging">
	<xs:complexType>
		<xs:sequence>
			<xs:element minOccurs="0" name="BidirectionalChargingEnabled" type="HPXMLBoolean"/>
			<xs:element minOccurs="0" name="GridConnected" type="HPXMLBoolean"/>
			<xs:element minOccurs="0" name="BuildingConnected" type="HPXMLBoolean"/>
		</xs:sequence>
	</xs:complexType>
</xs:element>
```

And apply it both the `ElectricVehicleCharger` and `BatteryElectricVehicle`. This could allow bidirectional charging to be generally enabled but not necessarily connected to a building/grid.

Reply:
I have a lot of the same questions as @shorowit about whether and how this should be included on the car or the EVSE or what. I feel like we're missing a lot of detail on this nascent technology. Is there a use case we're trying to support with this? If so, let's let that inform what it should say. If not, I say we leave it off for now until we have more information.